.pipelines/reusable-build.yml

```diff
@@ -0,0 +1,212 @@
+# SPDX-License-Identifier: MIT
+
+# This workflow performs a build of the project and uploads the result as a build artifact.
+
+parameters:
+  # Name associated with the output of this build.
+  - name: build_artifact
+    type: string
+  # Name of the job
+  - name: name
+    type: string
+  # Additional options passed to msbuild.
+  - name: build_options
+    type: string
+    default: ''
+  - name: generate_release_package
+    type: boolean
+    default: false
+  - name: build_codeql
+    type: boolean
+    default: false
+  - name: build_msi
+    type: boolean
+    default: false
+  - name: build_nuget
+    type: boolean
+    default: false
+  - name: cxx_flags
+    type: string
+    default: ''
+  - name: ld_flags
+    type: string
+    default: ''
+  - name: configurations
+    type: object
+    default: ["Debug", "Release"]
+
+
+jobs:
+  - job: '${{parameters.name}}'
+    timeoutInMinutes: 90
+
+    strategy:
+      matrix:
+        ${{ each configuration in parameters.configurations }}:
+          ${{ configuration }}:
+            buildConfiguration: ${{ configuration }}
+
+    pool:
+      vmImage: 'windows-2022'
+      type: windows
+
+    variables:
+      # Path to the solution file relative to the root of the project.
+      SOLUTION_FILE_PATH: ebpf-for-windows.sln
+      BUILD_ARTIFACT_NAME: ${{parameters.build_artifact}}
+      BUILD_CONFIGURATION: $(buildConfiguration)
+      BUILD_PLATFORM: x64
+      BUILD_OPTIONS: ${{parameters.build_options}}
+      CXX_FLAGS: ${{parameters.cxx_flags}}
+      LD_FLAGS: ${{parameters.ld_flags}}
+      MSBUILD_PATH: 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64'
+      GDN_CODESIGN_TARGETDIRECTORY: '$(Build.SourcesDirectory)/$(BUILD_PLATFORM)/$(buildConfiguration)'
+
+    steps:
+      - script: |
+          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+          echo "##vso[task.setvariable variable=msvc_tools_path;isOutput=true]%VCToolsInstallDir%"
+          echo "##vso[task.setvariable variable=msvc_tools_version;isOutput=true]%VCToolsVersion%"
+        name: msvc_variables
+        displayName: 'Set MSVC Environment Variables'
+
+      - powershell: |
+          $value = "true"
+          Write-Host "##vso[task.setvariable variable=ASAN_WIN_CONTINUE_ON_INTERCEPTION_FAILURE;]$value"
+        workingDirectory: $(Build.SourcesDirectory)
+        displayName: 'Set ASAN_WIN_CONTINUE_ON_INTERCEPTION_FAILURE'
+
+      - checkout: self
+        displayName: 'Checkout Repo'
+        submodules: 'recursive'
+        fetchDepth: 0
+
+      - bash: |
+          echo "*** All environment variables ***"
+          env | sort
+        workingDirectory: $(Build.SourcesDirectory)
+        displayName: 'Dump Environment variables'
+
+      - bash: |
+          choco install -y llvm --version 11.0.1 --allow-downgrade
+          echo '##vso[task.prependpath]C:\Program Files\LLVM\bin'
```

Comment:
This also hard codes the path to LLVM and should really use a better way to get the path.